app/provider_details/rest.py

```diff
@@ -29,19 +29,37 @@ def get_provider_by_id(provider_details_id):
     return jsonify(provider_details=data)
 
 
+@provider_details.route('/<uuid:provider_details_id>/versions', methods=['GET'])
+def get_provider_versions(provider_details_id):
+    versions = dao_get_provider_versions(provider_details_id)
+    data = provider_details_history_schema.dump(
+        versions,
+        many=True
+    ).data
+    return jsonify(data=data)
+
+
 @provider_details.route('/<uuid:provider_details_id>', methods=['POST'])
 def update_provider_details(provider_details_id):
-    fetched_provider_details = get_provider_details_by_id(provider_details_id)
+    valid_keys = {'priority', 'created_by', 'active'}
+    invalid_keys = request.get_json().keys() - valid_keys
 
-    current_data = dict(provider_details_schema.dump(fetched_provider_details).data.items())
-    current_data.update(request.get_json())
-    update_dict = provider_details_schema.load(current_data).data
-
-    invalid_keys = {'identifier', 'version', 'updated_at'} & set(key for key in request.get_json().keys())
     if invalid_keys:
         message = "Not permitted to be updated"
         errors = {key: [message] for key in invalid_keys}
         raise InvalidRequest(errors, status_code=400)
 
-    dao_update_provider_details(update_dict)
-    return jsonify(provider_details=provider_details_schema.dump(fetched_provider_details).data), 200
+    provider = get_provider_details_by_id(provider_details_id)
+    req_json = request.get_json()
```

Comment:
Small thing, but you could set this variable above line 45 to avoid calling request.get_json() twice.